crypto/asn1/a_int.c

```diff
@@ -244,17 +242,25 @@ static int asn1_get_int64(int64_t *pr, const unsigned char *b, size_t blen,
     if (asn1_get_uint64(&r, b, blen) == 0)
         return 0;
     if (neg) {
-        if (r > ABS_INT64_MIN) {
+        if (r <= INT64_MAX) {
+            /* Most significant bit is guaranteed to be clear, negation
+             * is guaranteed to be meaningful in platform-neutral sense. */
+            *pr = -(int64_t)r;
+        } else if (r == ABS_INT64_MIN) {
+            /* This never happens if INT64_MAX == ABS_INT64_MIN, e.g.
+             * on ones'-complement system. */
+            *pr = (int64_t)(0 - r);
```

Comment:
Yes, the hypothetical was that 0's type would be so wide that r would be promoted to a (wider-than-64-bit) signed type.  All values of type smaller than int are promoted to int at basically every opportunity, so the '0' literal is type int (as opposed to char), and it would take a (presumably) 128-bit 'int' to achieve this hypothetical, so I don't care to worry about the hypothetical.